src/metabase/serialization/dump.clj

```diff
@@ -0,0 +1,294 @@
+                       qp.util/query->source-card-id
+                       Card
+                       fully-qualified-name)
+              (some->> card
+                       :collection_id
+                       Collection
+                       fully-qualified-name)
+              "/collections")
+          (:name card)))
+
+(defmethod fully-qualified-name nil
+  [_]
+  nil)
+
+(def ^:private SegmentOrMetric
+  [(s/one (s/constrained su/KeywordOrString
+                         (comp #{:metric :segment} qp.util/normalize-token))
+          "head")
+   (s/cond-pre s/Int su/KeywordOrString)])
+
+(def ^{:arglists '([form])} entity-reference?
+  "Is given form an MBQL entity reference?"
+  (some-fn field-reference? (complement (s/checker SegmentOrMetric))))
+
+(defn- entity-reference->fully-qualified-name
+  [[op & args :as entity-reference]]
+  (case (qp.util/normalize-token op)
+    :metric        [op (fully-qualified-name (Metric (first args)))]
+    :segment       [op (fully-qualified-name (Segment (first args)))]
+    :field-id      [op (if (string? (first args))
+                         (first args)
+                         (fully-qualified-name (Field (first args))))]
+    :fk->          (into [op]
+                         (for [arg args]
+                           (if (number? arg)
+                             (Field arg)
+                             (entity-reference->fully-qualified-name arg))))
+    :field-literal entity-reference))
+
+(defn- humanize-entity-references
+  [entity]
+  (walk/postwalk
+   (fn [form]
+     (cond
+       (entity-reference? form)
+       (entity-reference->fully-qualified-name form)
+
+       (map? form)
+       (let [id->fully-qualified-name (fn [entity-id model]
+                                        (if (string? entity-id)
+                                          entity-id
+                                          (fully-qualified-name (model entity-id))))]
+         (-> form
+             (u/update-when :database (fn [db]
+                                        (if (= db -1337)
+                                          "database/virtual"
+                                          (id->fully-qualified-name db Database))))
+             (u/update-when :card_id id->fully-qualified-name Card)
+             (u/update-when :source-table (fn [source-table]
+                                            (if (and (string? source-table)
+                                                     (str/starts-with? source-table "card__"))
+                                              (-> source-table
+                                                  (str/split #"__")
+                                                  second
+                                                  Integer/parseInt
+                                                  (id->fully-qualified-name Card))
+                                              (id->fully-qualified-name source-table Table))))))
+
```

Comment:
probably would be a goo idea to split this ~humanize~ fully-qualify fn up into smaller pieces. Move the map part into a ~humanize-map~ `fully-qualify-map` fn for example. Clojure's already dense and it gets hard to look at 30-line functions like this. Plus something like `fully-qualify-map` would be easier to test or play with in the REPL
Might even